expr/interval_compound_literal.go

```diff
@@ -0,0 +1,154 @@
+package expr
+
+import (
+	"errors"
+
+	"github.com/substrait-io/substrait-go/proto"
+	"github.com/substrait-io/substrait-go/types"
+)
+
+type intervalDateParts struct {
+	years      int32
+	months     int32
+	days       int32
+	seconds    int32
+	subSeconds int64
+}
+
+type intervalDatePartsOptions func(parts *intervalDateParts)
+
+func WithIntervalCompoundYears(years int32) func(*intervalDateParts) {
+	return func(idp *intervalDateParts) {
+		idp.years = years
+	}
+}
+
+func WithIntervalCompoundMonths(months int32) func(*intervalDateParts) {
+	return func(idp *intervalDateParts) {
+		idp.months = months
+	}
+}
+
+func WithIntervalCompoundDays(days int32) func(*intervalDateParts) {
+	return func(idp *intervalDateParts) {
+		idp.days = days
+	}
+}
+
+func WithIntervalCompoundSeconds(seconds int32) func(*intervalDateParts) {
+	return func(idp *intervalDateParts) {
+		idp.seconds = seconds
+	}
+}
+
+func WithIntervalCompoundSubSeconds(subSeconds int64) func(*intervalDateParts) {
```

Comment:
not a fan of this. think it should probably be WithMilliseconds, WithMicroseconds, WithNanoseconds.

Reply:
fixed it